cloudinit/distros/parsers/ifconfig.py

```diff
@@ -87,10 +88,11 @@ class Ifconfig:
     """
 
     def __init__(self):
-        self._ifs = {}
+        self._ifs_by_name = {}
+        self._ifs_by_mac = {}
 
     @lru_cache()
-    def parse(self, text: str) -> Dict[str, Ifstate]:
+    def parse(self, text: str) -> Dict[str, Union[Ifstate, List[Ifstate]]]:
```

Comment:
I guess the type change is not needed as we do not return a list, right?

```suggestion
    def parse(self, text: str) -> Dict[str, Ifstate]:
```

Reply:
we do return a str: List type dict in the case of duplicate mac addresses tho

Reply:
Indeed, still booting up from holidays :)